woocommerce/woocommerce
# Migrate Table component to TS

### All Submissions:






### Changes proposed in this Pull Request:



Closes #35379 .

Migrate `Table` component to TS.






### How to test the changes in this Pull Request:

1. Run `pnpm --filter=@woocommerce/storybook run storybook`
2. Go to http://localhost:6007/
3. Test `Table`, `TableCard`, `TablePlaceholder`, `TableSummaryPlaceholder`, and `EmptyTable`

Extra point: 

1. Use this branch locally and make a JN site.
2. Use [wc-smooth-generator](https://github.com/woocommerce/wc-smooth-generator) to feed some orders to both sites.
3. Compare `Analytics -> Products` and `Analytics -> Orders` side by side and see if there are any differences.



### Other information:




### FOR PR REVIEWER ONLY:
